test/testcafe/framework/page-objects/ChallengeOktaVerifyPushPageObject.js

```diff
@@ -1,86 +1,102 @@
 import { Selector, userVariables } from 'testcafe';
 import ChallengeFactorPageObject from './ChallengeFactorPageObject';
-import { within } from '@testing-library/testcafe';
 
 const FORM_INFOBOX_WARNING = '.okta-form-infobox-warning';
+const FORM_INFOBOX_ERROR = '[data-se="o-form-error-container"] [data-se="callout"]';
 const RESEND_NUMBER_CHALLENGE_BUTTON = '.okta-form-infobox-warning .resend-number-challenge';
 const FORM_INFOBOX_ERROR_TITLE = '[data-se="o-form-error-container"] [data-se="callout"] > h3';
 const FORM_SELECTOR = '.okta-verify-send-push-form';
 const FORM_SELECTOR_V3 = '.mfa-okta-verify';
 const AUTO_CHALLENGE_CHECKBOX_SELECTOR = '[name$="autoChallenge"]';
 const AUTO_CHALLENGE_CHECKBOX_LABEL_SELECTOR = '[data-se-for-name$="autoChallenge"]';
 const FACTOR_BEACON = '.auth-beacon.auth-beacon-factor';
-const RESEND_PUSH_NOTIFICATION_TEXT = 'Resend push notification';
+const INFOBOX_WARNING_V3 = '[role="alert"]';
 export default class ChallengeOktaVerifyPushPageObject extends ChallengeFactorPageObject {
   constructor(t) {
     super(t);
-    this.beacon = new Selector('.beacon-container');
+    if (userVariables.v3) {
+      this.beacon = new Selector('[data-se="factor-beacon"]');
+    } else {
+      this.beacon = new Selector('.beacon-container');
+    }
   }
 
   getPushButton() {
+    if (userVariables.v3) {
+      return this.form.getButton('Push notification sent');
+    }
     return this.form.getElement('.send-push');
   }
 
+  async isPushButtonDisabled() {
+    const pushBtnClass = await this.getPushButton().getAttribute('class');
+    return pushBtnClass.includes('disabled');
+  }
+
   getA11ySpan() {
     return this.form.getElement('.accessibility-text');
   }
 
   getResendPushButton() {
-    if (userVariables.v3) {
-      return this.form.getButton(RESEND_PUSH_NOTIFICATION_TEXT);
-    }
     return this.form.getElement('.button-primary');
   }
 
   clickResendPushButton() {
-    if (userVariables.v3) {
-      return this.form.clickSaveButton(RESEND_PUSH_NOTIFICATION_TEXT);
-    }
     return this.form.clickSaveButton();
   }
 
-  clickResendNumberChallenge() {
+  async clickResendNumberChallenge() {
+    if (userVariables.v3) {
+      return this.form.clickElement('a');
+    }
     return this.form.clickElement(RESEND_NUMBER_CHALLENGE_BUTTON);
```

Comment:
would `this.t.click(this.form.getLink('resend the push notification'));` work here instead?  To make it a little more semantic